### DISCUSSION TITLE
Long Term Goals and Vision

### DISCUSSION THREAD

**Discussion by stephaje** [sbondCo/Watcharr#356] [General]
Hey @IRHM,

I was wanting to ask what the long term goals of this project are and what role you see this application taking on in the *arr family.

I think the following questions would shed some light on the project scope and long-term growth directions.

### Users
1. Who are the users?
    - A group who shares a Plex/Jellyfin instance?
    - A group who sources their own media and just shares ratings?
    - Does everyone in the instance know each other?
2. What are the users here to do?
    - Track things they've watched, obviously
    - See the watch and ratings history of their friends
    - Discover new content
    - Request new content they discover?
### Discovery
3. Users can discover new content by...
    - New/upcoming
    - What is popular amongst their friends?
    - Things matching their taste as determined by their ratings?
       - Including older content they might have missed?
       - Leveraging things they've marked as Dropped or "Not Interested"?
4. Users can request new content to be made available...
    - Without approval?
    - With server owner approval?
    - Not at all; out of scope
### Place in the Stack
5. This is meant to work alongside...
    - Sonarr/Radarr
    - Twitch
    - Plex/Jellyfin?
    - Steam?
    - Overseerr/Jellyseerr?
6. Will this eventually be able to fully replace Overseerr/Jellyseerr?

I ask because I am particularly interested in discovery/recommendation functionality, but don't want to pigeonhole my ideas where they don't belong.

Cheers,
John

**Comment by IRHM** [comment]
Hi @stephaje, thanks for taking the time to create a discussion.

Watcharr probably doesn't qualify to be a part of the *arr family, I just thought the name was nice :() and if you watch movies/tv, you might want to track what you watch too.

**Users**

I personally enjoy adding to my list after I watch something, so I'd like to think anyone who likes to do the same would be a target user.

Following friends and viewing their ratings adds to this because you can quickly share your favourite movies/series with anyone, and they can check back whenever they want to see what you are watching now to get their own recommendations that way. I get a lot of show recommendations from friends/family, I figure if you are close to someone, you probably share tastes in media.

So whether you use plex/jellyfin, only watch movies in the cinema or only watch on Netflix/other apps, Watcharr can be for you.

There is basic support for requesting content from Radarr/Sonarr that might be extended upon in the future (#211), though not sure if anyone would use it.

**Discovery**

The current ways to discover content is by checking your friends list, checking the Discovery page or by clicking into movies/tv to view similar content.

There's nothing crazy advanced that can take into account your rating/status on previous watched content; not too sure if it would be possible or how to go about doing that (also not sure if systems like that ever produce good recommendations?).

At the moment, only admins can request content to be available on Radarr/Sonarr, but permissions are something that I want to extend upon in the future.

**Place In The Stack**

Works alongside any platform as long as it is centered around Movies/Tv/Anime/Games.

I'm not sure if Watcharr could ever fully replace Overseerr/Jellyseerr, but in the future it could appeal to users (admins) who like a simpler option, since when content requesting is finished, you just give the users that you allow to request a single permission and they will be able to do so whenever they'd like to.

On Jellyseer, I just set it to auto accept any content, since I trust the users, so the feature on Watcharr would be simple/basic as if it were set like that on Jellyseerr:
    
User want content -> user click request -> user sees download progress -> user wait until done.

---

P.S. Don't worry about pigeonholing your ideas! Feel free to suggest anything you'd like to see become a feature.

**Comment by stephaje** [comment]
> Watcharr probably doesn't qualify to be a part of the *arr family, I just thought the name was nice :()

I don't think it qualifies yet, but I think it can get there with all of the other things discussed. And if that is an acceptable goal, then I'm happy to help out.

> not too sure if it would be possible or how to go about doing that (also not sure if systems like that ever produce good recommendations?).

Recommender systems I think often come up short because they either lack the depth of data needed to actually see patterns or they lack the justification (ROI) for the amount of resources to run a competent system. For the first point, I think a ratings database is a great place to curate the quantity of necessary data. For the second point, there is full control over the server code and it can do whatever the code supports.

Personalized recommendations has been something I've been working on for a while now, but as a back-end dev, it's ugly and hard to expand to my friends. With my large set of pass/fail ratings on movies, I'm very happy with my code's ability to alert me when new media comes out that it thinks fits my tastes. I think a reliable system is possible if people are willing to provide the quantity of data necessary.

> Works alongside any platform as long as it is centered around Movies/Tv/Anime/Games.

Nice.

----------------------------------------------------

If you think a robust recommendation system is something that would be cool to have, I'd love to help it become a part of the stack. And if that's a bit too far, then maybe there could just be an API for fetching a user's data that could then be pulled into another system.

Beyond that, I think there are other things that could bring a lot of value, such as OAuth/Plex/Jellyfin login support, watch history driven review prompting, and a few other things that I also wouldn't mind taking a look into. Let me know if you'd like me to start up individual idea discussions, write them up as issues/feature requests, enumerate them here, or if you'd like to discuss further on another platform.

**Reply by IRHM** [reply]
All sounds good @stephaje, do feel free to make separate issues on each feature.

Robust recommendations does sound good, as long as the code itself can stay pretty modular. We can definitely look more into it.

I'd love to see your implementation of this feature in the application you mention, if its available. May give me a better insight as to how this would work.

p.s. we can chat elsewhere if its easier than comments on issues, let me know, I have Matrix.
